### PR TITLE
Refine ShaderInputs::fixupUses()

### PR DESCRIPTION
**Pull request by xuechen417**
Re-struct the control flow in the inner loop of `fixupUses`.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4174570accc1174a811542544b7ced7a6cbcf65a
<details><summary>CTS tests (Failed: 0/211523)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70507 (55.7%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31259/70507 (44.3%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70507 (56.0%)</li>
<li>Failed: 0/70507 (0.0%)</li>
<li>Not Supported: 31009/70507 (44.0%)</li><li>Warnings: 0/70507 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70509 (51.5%)</li>
<li>Failed: 0/70509 (0.0%)</li>
<li>Not Supported: 34164/70509 (48.5%)</li><li>Warnings: 0/70509 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
Thanks, LGTM

File: lgc/patch/ShaderInputs.cpp [hunk 1]
```diff
@@ -333,46 +333,53 @@ void ShaderInputs::fixupUses(Module &module, PipelineState *pipelineState) {
...
-        for (Instruction *&call : inputUsage->users) {
-          if (call && call->getFunction() == &func) {
-            call->replaceAllUsesWith(arg);
-            call->eraseFromParent();
-            call = nullptr;
-          }
+      if (!inputUsage)
+        continue;
+      Value *value = nullptr;
+      {
```
Comment (tsymalla) [inline lgc/patch/ShaderInputs.cpp:339]: What is the benefit of putting this in a separate scope?

Comment (xuechen417) [reply lgc/patch/ShaderInputs.cpp:339]: It is going to serve an internal change.

**Pull request merged**